patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -2429,6 +2429,34 @@
  				UILinkPointNavigator.Shortcuts.CRAFT_CurrentRecipeBig = -1;
  				UILinkPointNavigator.Shortcuts.CRAFT_CurrentRecipeSmall = -1;
  				if (numAvailableRecipes > 0)
+@@ -30626,7 +_,7 @@
+ 				if (numAvailableRecipes > 0) {
+ 					UILinkPointNavigator.Shortcuts.CRAFT_CurrentRecipeBig = -1;
+ 					UILinkPointNavigator.Shortcuts.CRAFT_CurrentRecipeSmall = -1;
+-					for (int num72 = 0; num72 < Recipe.maxRequirements; num72++) {
++					for (int num72 = 0; num72 < recipe[availableRecipe[focusRecipe]].requiredItem.Count; num72++) {
+ 						if (recipe[availableRecipe[focusRecipe]].requiredItem[num72].type == 0) {
+ 							UILinkPointNavigator.Shortcuts.CRAFT_CurrentIngridientsCount = num72 + 1;
+ 							break;
+@@ -30671,7 +_,8 @@
+ 						UILinkPointNavigator.Shortcuts.CRAFT_CurrentRecipeSmall = 1 + num72;
+ 						Microsoft.Xna.Framework.Color color5 = inventoryBack;
+ 						inventoryBack = new Microsoft.Xna.Framework.Color((byte)num75, (byte)num75, (byte)num75, (byte)num75);
+-						ItemSlot.Draw(spriteBatch, ref recipe[availableRecipe[focusRecipe]].requiredItem[num72], 22, new Vector2(num73, num74));
++						Item tempItem = recipe[availableRecipe[focusRecipe]].requiredItem[num72];
++						ItemSlot.Draw(spriteBatch, ref tempItem, 22, new Vector2(num73, num74));
+ 						inventoryBack = color5;
```

Comment:
Shouldn't you set the list entry's value to tempitem here?

Reply:
this is just for drawing the crafting menu, the actual item in the slot won't change